### PR TITLE
feat: port create_table to the async python API and the remote rust API

### PR DESCRIPTION
**Pull request by westonpace**
I've also started `ASYNC_MIGRATION.MD` to keep track of the breaking changes from sync to async python.

### CODE REVIEW

File: python/python/lancedb/table.py [hunk 1]
```diff
@@ -1780,3 +1782,715 @@ def _sanitize_nans(data, fill_value, on_bad_vectors, vec_arr, vector_column_name
...
+    Can append new data with [Table.add()][lancedb.table.Table.add].
+
+    >>> table.add([{"vector": [0.5, 1.3], "b": 4}])
+
+    Can query the table with [Table.search][lancedb.table.Table.search].
+
+    >>> table.search([0.4, 0.4]).select(["b"]).to_pandas()
+       b      vector  _distance
+    0  4  [0.5, 1.3]       0.82
+    1  2  [1.1, 1.2]       1.13
```
[the remark this answers is not in the document]
Comment (wjones127) [reply python/python/lancedb/table.py:1818]: Or is this left as a TODO?

File: python/src/connection.rs [hunk 2]
```diff
@@ -33,6 +34,65 @@ impl Connection {
...
+        let inner = self_.inner.clone();
+
+        let mode = if mode == "create" {
+            CreateTableMode::Create
+        } else if mode == "overwrite" {
+            CreateTableMode::Overwrite
+        } else {
+            // TODO: allow users to pass in open options
+            CreateTableMode::exist_ok(|builder| builder)
+        };
```
Comment (wjones127) [inline python/src/connection.rs:53]: nit: seems like a good use for `match`?
```suggestion
        match mode {
            "create" => CreateTableMode::Create,
            "overwrite" => CreateTableMode::Overwrite,
            _ => CreateTableMode::exist_ok(|builder| builder)
        }
```

File: rust/lancedb/src/remote/db.rs [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+        // to make sure we don't block the tokio runtime if the source is slow.
+        let data_buffer = spawn_blocking(move || batches_to_ipc_bytes(data))
+            .await
+            .unwrap()?;
+
+        self.client
+            .post(&format!("/v1/table/{}/create", options.name))
+            .body(data_buffer)
+            .header(CONTENT_TYPE, ARROW_STREAM_CONTENT_TYPE)
+            .header("x-request-id", "na")
```
Comment (wjones127) [inline rust/lancedb/src/remote/db.rs:81]: This is for tracing purposes?

File: nodejs/lancedb/native.d.ts [hunk 4]
```diff
@@ -117,7 +117,7 @@ export class Table {
   /** Return Schema as empty Arrow IPC file. */
   schema(): Promise<Buffer>
   add(buf: Buffer): Promise<void>
-  countRows(filter?: string | undefined | null): Promise<bigint>
```
[the remark this answers is not in the document]
Comment (westonpace) [reply nodejs/lancedb/native.d.ts:120]: Yep.  Removed.

File: python/python/lancedb/table.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (westonpace) [reply python/python/lancedb/table.py:1818]: I'm going to leave this as a TODO.  My rationale is that some of these examples need other capabilities (e.g. this one needs `add` and so it'll be easier to do the doc tests all at once at the end.

https://github.com/lancedb/lancedb/issues/1044

File: python/src/connection.rs (hunk 2 above)
Comment (westonpace) [reply python/src/connection.rs]: Switched to match and put in a helper function to share with `create_empty_table`.  I also updated the node version to behave more similarly to the python version.

File: rust/lancedb/src/remote/db.rs (hunk 3 above)
Comment (wjones127) [reply rust/lancedb/src/remote/db.rs:81]: Seems worth a comment.

**Review by wjones127 — APPROVED** [review]

**Pull request merged**